Scala.js 1.4.0 upgrade broke nightly CI (test_sbt)

Nightly CI has failed ever since #11073 was merged:

```
[info] [info] Fast optimizing /tmp/sbt_11b8a8d5/target/scala-3.0.0-RC1/sbt_11b8a8d5-fastopt
[info] [error] org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.4 (supported up to: 1.3): /tmp/sbt_11b8a8d5/target/scala-3.0.0-RC1/classes/test/Main.sjsir
[info] [error] You may need to upgrade the Scala.js sbt plugin to version 1.4 or later.
[info] [error] 	at org.scalajs.sbtplugin.ScalaJSPluginInternal$.enhanceIRVersionNotSupportedException(ScalaJSPluginInternal.scala:83)
[info] [error] 	at org.scalajs.sbtplugin.ScalaJSPluginInternal$.$anonfun$scalaJSStageSettings$18(ScalaJSPluginInternal.scala:225)
[info] [error] 	at sbt.util.FileFunction$.$anonfun$cached$1(FileFunction.scala:80)
[info] [error] 	at sbt.util.FileFunction$.$anonfun$cached$4(FileFunction.scala:153)
[info] [error] 	at sbt.util.Difference.apply(Tracked.scala:414)
[info] [error] 	at sbt.util.Difference.apply(Tracked.scala:394)
[info] [error] 	at sbt.util.FileFunction$.$anonfun$cached$3(FileFunction.scala:149)
[info] [error] 	at sbt.util.Difference.apply(Tracked.scala:414)
[info] [error] 	at sbt.util.Difference.apply(Tracked.scala:389)
[info] [error] 	at sbt.util.FileFunction$.$anonfun$cached$2(FileFunction.scala:148)
[info] [error] 	at org.scalajs.sbtplugin.ScalaJSPluginInternal$.$anonfun$scalaJSStageSettings$17(ScalaJSPluginInternal.scala:237)
[info] [error] 	at sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info] [error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[info] [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[info] [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[info] [error] 	at sbt.Execute.work(Execute.scala:291)
[info] [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[info] [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:265)
[info] [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:64)
[info] [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] [error] 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info] [error] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info] [error] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[info] [error] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[info] [error] 	at java.lang.Thread.run(Thread.java:748)
[info] [error] Caused by: org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.4 (supported up to: 1.3): /tmp/sbt_11b8a8d5/target/scala-3.0.0-RC1/classes/test/Main.sjsir
[info] [error] 	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:62)
[info] [error] 	at org.scalajs.linker.interface.unstable.IRFileImpl$$anonfun$withPathExceptionContext$1.applyOrElse(IRFileImpl.scala:58)
[info] [error] 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
[info] [error] 	at scala.util.Failure.recover(Try.scala:234)
[info] [error] 	at scala.concurrent.Future.$anonfun$recover$1(Future.scala:395)
[info] [error] 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[info] [error] 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[info] [error] 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[info] [error] 	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
[info] [error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[info] [error] 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[info] [error] 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[info] [error] 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
[info] [error] Caused by: org.scalajs.ir.IRVersionNotSupportedException: This version (1.4) of Scala.js IR is not supported. Supported versions are up to 1.3
[info] [error] 	at org.scalajs.ir.VersionChecks.checkSupported(ScalaJSVersions.scala:77)
[info] [error] 	at org.scalajs.ir.Serializers$Deserializer.readHeader(Serializers.scala:994)
[info] [error] 	at org.scalajs.ir.Serializers$Deserializer.deserializeEntryPointsInfo(Serializers.scala:953)
[info] [error] 	at org.scalajs.ir.Serializers$.$anonfun$deserializeEntryPointsInfo$1(Serializers.scala:52)
[info] [error] 	at org.scalajs.ir.Serializers$.withBigEndian(Serializers.scala:67)
[info] [error] 	at org.scalajs.ir.Serializers$.deserializeEntryPointsInfo(Serializers.scala:52)
[info] [error] 	at org.scalajs.linker.PathIRFile$PathIRFileImpl.$anonfun$entryPointsInfo$1(PathIRFile.scala:39)
[info] [error] 	at scala.util.Success.$anonfun$map$1(Try.scala:255)
[info] [error] 	at scala.util.Success.map(Try.scala:213)
[info] [error] 	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
[info] [error] 	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
[info] [error] 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[info] [error] 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[info] [error] 	at java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1402)
[info] [error] 	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
[info] [error] 	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
[info] [error] 	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
[info] [error] 	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
[info] [error] (Compile / fastLinkJS) org.scalajs.ir.IRVersionNotSupportedException: Failed to deserialize a file compiled with Scala.js 1.4 (supported up to: 1.3): /tmp/sbt_11b8a8d5/target/scala-3.0.0-RC1/classes/test/Main.sjsir
[info] [error] You may need to upgrade the Scala.js sbt plugin to version 1.4 or later.
[info] [error] Total time: 2 s, completed Jan 14, 2021 4:53:05 AM
Error:  x scalajs/basic 
Error:   Cause of test exception: {line 1}  Command failed: run failed
```

https://github.com/lampepfl/dotty/runs/1699561832?check_suite_focus=true

--- a/project/Build.scala
+++ b/project/Build.scala
@@ -1256,6 +1256,7 @@ object Build {
       scriptedLaunchOpts ++= Seq(
         "-Dplugin.version=" + version.value,
         "-Dplugin.scalaVersion=" + dottyVersion,
+        "-Dplugin.scalaJSVersion=" + scalaJSVersion,
         "-Dsbt.boot.directory=" + ((baseDirectory in ThisBuild).value / ".sbt-scripted").getAbsolutePath // Workaround sbt/sbt#3469
       ),
       // Pass along ivy home and repositories settings to sbt instances run from the tests

--- a/sbt-dotty/sbt-test/scalajs/basic/project/plugins.sbt
+++ b/sbt-dotty/sbt-test/scalajs/basic/project/plugins.sbt
@@ -1,2 +1,2 @@
 addSbtPlugin("ch.epfl.lamp" % "sbt-dotty" % sys.props("plugin.version"))
-addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.3.0")
+addSbtPlugin("org.scala-js" % "sbt-scalajs" % sys.props("plugin.scalaJSVersion"))

